core-bundle/src/Cron/PurgeOptInTokensCron.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Cron;
+
+use Contao\CoreBundle\Monolog\ContaoContext;
+use Contao\CoreBundle\OptIn\OptIn;
+use Contao\CoreBundle\ServiceAnnotation\CronJob;
+use Doctrine\DBAL\Connection;
+use Psr\Log\LoggerInterface;
+
+/**
+ * @CronJob("hourly")
+ */
+class PurgeOptInTokensCron
+{
+    /**
+     * @var OptIn
+     */
+    private $optIn;
+
+    /**
+     * @var Connection
+     */
+    private $database;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    public function __construct(OptIn $optIn, Connection $database, LoggerInterface $logger = null)
+    {
+        $this->optIn = $optIn;
+        $this->logger = $logger;
+        $this->database = $database;
+    }
+
+    public function __invoke(): void
+    {
+        $count = (int) $this->database->fetchOne('SELECT COUNT(*) FROM tl_opt_in');
+
+        $this->optIn->purgeTokens();
+
+        $count -= (int) $this->database->fetchOne('SELECT COUNT(*) FROM tl_opt_in');
```

Comment:
I don't like this. Is it really necessary to know how many tokens have been purged? If so, the `purgeTokens()` method should probably return the value.

Reply:
> I don't like this. Is it really necessary to know how many tokens have been purged?

No, it's just nice to have. We did not have this before, so I am fine with removing it.

> If so, the purgeTokens() method should probably return the value.

As already mentioned in https://github.com/contao/contao/pull/3711#discussion_r75826067 we cannot change the return type as it is part of the interface.

Reply:
Then I am for dropping the `$count` variable entirely.

Reply:
Done.